test/e2e/cases/basic.go

```diff
@@ -32,15 +32,17 @@ var kubemci = "kubemci"
 // Creates a basic multicluster ingress and verifies that it works as expected.
 // TODO(nikhiljindal): Use ginkgo and gomega and possibly reuse k/k e2e framework.
 func RunBasicCreateTest() {
-	project, kubeConfigPath, lbName, ipName, clients := setup()
+	project, kubeConfigPath, lbName, ipName, clients := setupBasicDependencies()
+	setupBasic(kubeConfigPath, ipName, clients)
 	defer func() {
-		cleanup(kubeConfigPath, lbName, ipName, clients)
+		cleanupBasic(kubeConfigPath, lbName, ipName, clients)
+		cleanupBasicDependencies(ipName)
 	}()
 	testHTTPIngress(project, kubeConfigPath, lbName)
 	testHTTPSIngress(project, kubeConfigPath, lbName, clients)
 }
 
-func setup() (project, kubeConfigPath, lbName, ipName string, clients map[string]kubeclient.Interface) {
+func setupBasicDependencies() (project, kubeConfigPath, lbName, ipName string, clients map[string]kubeclient.Interface) {
```

Comment:
Add a comment for what this does. Its not clear from the names setupBasicDependencies and setupBasic what they both do and what is the difference.

Maybe call this init() so that it is clear that all tests will first call this. And then they will call setupBasic or setupHCFromProbe or setupMyTest for custom setup.

Reply:
Apparently `init()` has special semantics in go, but I'll go for `initDeps()`